### PR TITLE
CXP-1375: Refactor useInfiniteAttribute to useInfiniteSourceAttribute

### PR DESCRIPTION
**Pull request by willy-ahva**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by eoliveauakeneo — APPROVED** [review]

File: components/catalogs/back/tests/Integration/Infrastructure/Controller/Internal/GetAttributesByTargetTypeAndTargetFormatActionTest.php [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            [],
+            [
+                'HTTP_X-Requested-With' => 'XMLHttpRequest',
+            ],
+        );
+
+        $response = $client->getResponse();
+        Assert::assertEquals(400, $response->getStatusCode());
+    }
+}
```
Comment (hogren) [inline components/catalogs/back/tests/Integration/Infrastructure/Controller/Internal/GetAttributesByTargetTypeAndTargetFormatActionTest.php:123]: Just to be sure, I see many error use cases are not tested here. Is it enough ?

**Review by hogren — APPROVED** [review]

File: components/catalogs/back/src/Infrastructure/Symfony/Resources/meta-schemas/product-0.0.4.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
+{
```
Comment (masacc) [inline components/catalogs/back/src/Infrastructure/Symfony/Resources/meta-schemas/product-0.0.4.json:1]: you also need to do a PR in pim-api-docs like this one: https://github.com/akeneo/pim-api-docs/pull/741

File: components/catalogs/back/tests/Integration/Infrastructure/Validation/ProductSchema/invalid/0.0.4-with-unsupported-format.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  "description": "Additional object properties are not allowed: format at #/properties/release_date",
+  "properties": {
+    "uuid": {
+      "title": "Product uuid",
+      "type": "string"
+    },
+    "release_date": {
+      "type": "string",
+      "format": "unknown-enum"
+    }
```
Comment (masacc) [inline components/catalogs/back/tests/Integration/Infrastructure/Validation/ProductSchema/invalid/0.0.4-with-unsupported-format.json:13]: missing another invalid use case:  `0.0.4-with-format-for-wrong-type.json`
```
    "release_date": {
      "type": "number",
      "format": "date-time"
    }
```

File: components/catalogs/back/tests/Integration/Infrastructure/Validation/ProductSchema/valid/0.0.4.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            "title": "Description",
+            "description": "Product description in raw HTML",
+            "type": "string"
+        },
+        "is_published": {
+            "type": "boolean"
+        },
+        "height": {
+            "type": "number"
+        }
```
Comment (masacc) [inline components/catalogs/back/tests/Integration/Infrastructure/Validation/ProductSchema/valid/0.0.4.json:26]: you must add a property with format:
```
    "release_date": {
      "type": "string",
      "format": "date-time"
    }
 ```

File: components/catalogs/back/tests/Unit/Service/TargetTypeConverterTest.php [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+    protected function setUp(): void
+    {
+        $this->targetTypeConverter = new TargetTypeConverter();
+    }
+
+    /**
+     * @dataProvider validConversionProvider
+     */
+    public function testItConvertTargetTypeToAttributeTypes(
```
Comment (masacc) [inline components/catalogs/back/tests/Unit/Service/TargetTypeConverterTest.php:27]:
```suggestion
    public function testItConvertsTargetTypeToAttributeTypes(
```
:smile:

File: components/catalogs/back/src/Application/Service/TargetTypeConverter.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+use Akeneo\Catalogs\Application\Exception\NoCompatibleAttributeTypeException;
+
+/**
+ * @copyright 2023 Akeneo SAS (https://www.akeneo.com)
+ * @license   https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class TargetTypeConverter
+{
+    private const PIM_ATTRIBUTE_TYPES = [
+        'string' => [
```
Comment (masacc) [inline components/catalogs/back/src/Application/Service/TargetTypeConverter.php:16]: You added `string+date-time` in your meta schema 0.0.4 and don't treat it here. Same for `boolean` and `number`. There is a lack of consistency here.

File: components/catalogs/back/tests/Unit/Service/TargetTypeConverterTest.php [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    public function validConversionProvider(): array
+    {
+        return [
+            'string' => [
+                'string',
+                '',
+                [
+                    'pim_catalog_text',
+                ]
+            ],
```
Comment (masacc) [inline components/catalogs/back/tests/Unit/Service/TargetTypeConverterTest.php:47]: some use cases declared in product schema 0.0.4 are missing (number, boolean, ...)

File: components/catalogs/back/src/Application/Exception/NoCompatibleAttributeTypeException.php [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+declare(strict_types=1);
+
+namespace Akeneo\Catalogs\Application\Exception;
+
+/**
+ * @copyright 2023 Akeneo SAS (https://www.akeneo.com)
+ * @license   https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+final class NoCompatibleAttributeTypeException extends \DomainException
+{
```
[the remark this answers is not in the document]
Comment (guillaumefer-akeneo) [reply components/catalogs/back/src/Application/Exception/NoCompatibleAttributeTypeException.php:12]: You are right. The intention behind this exception is to show that there is no compatible attribute type found. So I'll rename it `NoCompatibleAttributeTypeFoundException`.

File: components/catalogs/back/src/Application/Service/TargetTypeConverter.php (hunk 6 above)
Comment (guillaumefer-akeneo) [reply components/catalogs/back/src/Application/Service/TargetTypeConverter.php:16]: This PR is the foundation of the others PRs, the 'string+date-time' key will be added in the rework of this PR : https://github.com/akeneo/pim-community-dev/pull/18887

File: components/catalogs/back/tests/Integration/Infrastructure/Validation/ProductSchema/invalid/0.0.4-with-unsupported-format.json (hunk 3 above)
Comment (guillaumefer-akeneo) [reply components/catalogs/back/tests/Integration/Infrastructure/Validation/ProductSchema/invalid/0.0.4-with-unsupported-format.json:13]: Ok, it's added

File: components/catalogs/back/tests/Integration/Infrastructure/Validation/ProductSchema/valid/0.0.4.json (hunk 4 above)
Comment (guillaumefer-akeneo) [reply components/catalogs/back/tests/Integration/Infrastructure/Validation/ProductSchema/valid/0.0.4.json:26]: Not in this PR but in the next one : https://github.com/akeneo/pim-community-dev/pull/18887

File: components/catalogs/back/tests/Unit/Service/TargetTypeConverterTest.php (hunk 7 above)
Comment (guillaumefer-akeneo) [reply components/catalogs/back/tests/Unit/Service/TargetTypeConverterTest.php:47]: It's the foundation, use cases will be added in next PRs

**Pull request merged**